src/main/java/com/dumbdogdiner/stickyapi/common/nbt/NbtJsonAdapter.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2020-2021 DumbDogDiner <dumbdogdiner.com>. All rights reserved.
+ * Licensed under the MIT license, see LICENSE for more information...
+ */
+package com.dumbdogdiner.stickyapi.common.nbt;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonPrimitive;
+import lombok.experimental.UtilityClass;
+import org.jetbrains.annotations.NotNull;
+
+import java.text.MessageFormat;
+
+/**
+ * Utility class for easy creation of the appropriate {@link NbtTag} from an existing {@link JsonElement}
+ */
+@UtilityClass
+public final class NbtJsonAdapter {
+    /**
+     * Creates the appropriate tag from a given {@link JsonElement}, but never creates a {@link NbtJsonTag}
+     * @param jse The incoming {@link JsonElement} to convert
+     * @return the new {@link NbtTag}
+     */
+    public static NbtTag jsonToNbt(@NotNull JsonElement jse) {
+        if(jse.isJsonArray()){
+            return NbtListTag.fromJsonArray(jse.getAsJsonArray());
+        } else if(jse.isJsonObject()){
+            return NbtCompoundTag.fromJsonObject(jse.getAsJsonObject());
+        } else if(jse.isJsonPrimitive()){
+            JsonPrimitive primitive = jse.getAsJsonPrimitive();
+            if(primitive.isBoolean()){
+                return NbtBooleanTag.fromPrimitive(primitive);
+            } else if(primitive.isNumber()){
+                return NbtNumberTag.fromPrimitive(primitive);
+            } else if(primitive.isString()){
+                return NbtStringTag.fromPrimitive(primitive);
+            } else {
+                throw new UnsupportedOperationException("Illegal type of NBT primitive");
```

Comment:
if possible, please use a switch statement here, to make it more readable :)

Reply:
I really wish I could, but sadly I cannot think of a way to. You sorta can with kotlin I think, but I'm not fully sure

Reply:
then at least collapse it to something like this? There doesn't seem to be a `getType()` method sadly, so this is the only thing we can do to make it appear cleaner
```java
        if(jse.isJsonArray()){
            return NbtListTag.fromJsonArray(jse.getAsJsonArray());
        } else if(jse.isJsonObject()){
            return NbtCompoundTag.fromJsonObject(jse.getAsJsonObject());
        } else if(jse.isJsonPrimitive()){
            JsonPrimitive primitive = jse.getAsJsonPrimitive();
            if(primitive.isBoolean() || primitive.isNumber() || primitive.isString()) {
                return NbtBooleanTag.fromPrimitive(primitive);
            else {
                throw new UnsupportedOperationException("Illegal type of NBT primitive");
```